src/main/java/org/jabref/gui/openoffice/OOBibBase.java

```diff
@@ -68,22 +69,21 @@ public class OOBibBase {
 
     private final DialogService dialogService;
 
-    // Shall we add "Cited on pages: ..." to resolved bibliography entries?
-    private final boolean alwaysAddCitedOnPages; // TODO (see comment above)
+    private final boolean alwaysAddCitedOnPages;
 
     private final OOBibBaseConnect connection;
 
     private CSLCitationOOAdapter cslCitationOOAdapter;
 
-    public OOBibBase(Path loPath, DialogService dialogService)
+    public OOBibBase(Path loPath, DialogService dialogService, OpenOfficePreferences openOfficePreferences)
             throws
             BootstrapException,
             CreationException {
 
         this.dialogService = dialogService;
         this.connection = new OOBibBaseConnect(loPath, dialogService);
 
-        this.alwaysAddCitedOnPages = false;
+        this.alwaysAddCitedOnPages = openOfficePreferences.getAlwaysAddCitedOnPages();
```

Comment:
I wasn't sure here if I should inject the boolean value itself, the `OpenOfficePreferences` or the `PreferencesService`, I opted to stay closer to the real object while maintaining a layer of abstraction.

Let me know if this was appropriate for our standards or not.

Reply:
Yes this is fine. Especially since in the background the preferencs are observables you make sure that you are always getting the latest value